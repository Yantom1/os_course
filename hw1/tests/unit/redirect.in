ls
echo test_redirect> new
cat new
echo again_redirect> new
cat new
echo test_redirect>> new
cat new
showpid > mypid
cat mypid
showpid >> mypid
cat mypid
showpid > mypid
cat mypid
ls
sleep 100&
jobs > s
cat s
kill 1 > k
cat k
kill 10 > t
cat t
ls
ls > p
cat p
ls
ls > /dfgdfg/dfgdfg
ls >/tmp/smash/test/new
echo before
cat new
chmod 0 new
# Should fail
ls > new
rm new
ls
ls>new
cat new
ls
rm new
ls> new
cat new
ls
rm new
ls >new
cat new
mkdir test
ls > test/again
cat test/again
./my_sleep 6 > sl
!time.sleep(7)
cat sl
quit kill
