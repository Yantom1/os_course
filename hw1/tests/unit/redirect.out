smash> my_sleep
random1.txt
random2.txt
sub1
sub2
sub3
smash> smash> test_redirect
smash> smash> again_redirect
smash> smash> again_redirect
test_redirect
smash> smash> smash pid is 45027
smash> smash> smash pid is 45027
smash pid is 45027
smash> smash> smash pid is 45027
smash> mypid
my_sleep
new
random1.txt
random2.txt
sub1
sub2
sub3
smash> smash> smash> [1] sleep 100& : 45054 1 secs
smash> smash error: kill: invalid arguments
smash> smash> smash error: kill: invalid arguments
smash> smash> k
mypid
my_sleep
new
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> smash error: open failed: No such file or directory
smash> smash error: open failed: No such file or directory
smash> before
smash> again_redirect
test_redirect
smash> smash> smash error: open failed: Permission denied
smash> smash> k
mypid
my_sleep
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> smash> smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> smash> smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
smash> smash> smash> k
mypid
my_sleep
new
p
random1.txt
random2.txt
s
sub1
sub2
sub3
t
test
smash> smash> 0
2
4
Done: 6
smash> smash: sending SIGKILL signal to 1 jobs:
45054: sleep 100&
